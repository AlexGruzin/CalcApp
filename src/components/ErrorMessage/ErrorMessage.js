import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './ErrorMessage.styles.less';

const ErrorMessage = ({ className, children }) => (
  <div className={classNames(className, 'error-wrp')}>
    <img src="/svg/angle-left.svg" className="error-wrp__icon" alt="error icon" />
    <div className="error-wrp__content">{children}</div>
  </div>
);

ErrorMessage.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array, PropTypes.string]),
};

ErrorMessage.defaultProps = {
  className: '',
  children: null,
};

export default ErrorMessage;
