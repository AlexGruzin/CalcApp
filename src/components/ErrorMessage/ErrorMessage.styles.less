.error-wrp {
  display: flex;
  border: 1px solid var(--error-border-dark);
  border-radius: 7px;
  background: var(--error-background);
  padding: var(--space-x2);
}

.error-wrp__icon {
  display: flex;
  align-self: flex-start;
  margin: var(--space-x1);
  margin-right: var(--space-x3);
}

.error-wrp__content {
  margin: auto 0;
}