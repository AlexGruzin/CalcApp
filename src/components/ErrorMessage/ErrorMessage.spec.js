import React from 'react';
import { render } from '@testing-library/react';

import ErrorMessage from './ErrorMessage';

describe('ErrorMessage tests', () => {
  it('Check snapshots of ErrorMessage are equal.', () => {
    const { asFragment } = render(<ErrorMessage className="custom-error">{'Custom error'}</ErrorMessage>);
    expect(asFragment()).toMatchSnapshot();
  });
});
