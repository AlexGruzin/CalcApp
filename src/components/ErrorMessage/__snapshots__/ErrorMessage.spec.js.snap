// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage tests Check snapshots of ErrorMessage are equal. 1`] = `
<DocumentFragment>
  <div
    class="custom-error error-wrp"
  >
    <img
      alt="error icon"
      class="error-wrp__icon"
      src="/svg/angle-left.svg"
    />
    <div
      class="error-wrp__content"
    >
      Custom error
    </div>
  </div>
</DocumentFragment>
`;
